molecule_libvirt/playbooks/destroy.yml

```diff
@@ -0,0 +1,146 @@
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      when: (instance_list | json_query(jmesquery)) != []
+      vars:
+        jmesquery: "results[?item.name=='{{ item.name }}'].list_vms[?contains(@, '{{ item.name }}')]"
+
+    - name: Get libvirt vnet list
+      virt_net:
+        command: list_nets
+        uri: "{{ 'qemu+ssh://'+item.libvirt_user+'@'+item.libvirt_host+'/system' if item.libvirt_host is defined else libvirt_uri }}"
+      register: net_list
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+
+    - name: Create custom network dict
+      set_fact:
+        instance_and_net_list: "{{ net_list.results |
+        json_query('[].{name: item.name, network: list_nets}') }}"
+
+    - name: Stop molecule custom network
+      virt_net:
+        command: stop
+        name: "{{ molecule_network_name }}"
+        uri: "{{ 'qemu+ssh://'+item.libvirt_user+'@'+item.libvirt_host+'/system' if item.libvirt_host is defined else libvirt_uri }}"
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      when: item.bridge_name is not defined and
+        molecule_network_name in (instance_and_net_list |
+        json_query(jmesquery))
+      vars:
+        jmesquery: "[?name=='{{ item.name }}'].network[]"
+
+    - name: Destroy molecule network
+      virt_net:
+        command: destroy
+        name: "{{ molecule_network_name }}"
+        uri: "{{ 'qemu+ssh://'+item.libvirt_user+'@'+item.libvirt_host+'/system' if item.libvirt_host is defined else libvirt_uri }}"
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      when: item.bridge_name is not defined and
+        molecule_network_name in (instance_and_net_list |
+        json_query(jmesquery))
+      vars:
+        jmesquery: "[?name=='{{ item.name }}'].network[]"
+
+    - name: Undefine molecule network
+      virt_net:
+        command: undefine
+        name: "{{ molecule_network_name }}"
+        uri: "{{ 'qemu+ssh://'+item.libvirt_user+'@'+item.libvirt_host+'/system' if item.libvirt_host is defined else libvirt_uri }}"
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      when: item.bridge_name is not defined
+
+    - name: Remove vm image(s)
+      file:
+        path: "{{ libvirt_default_image_path }}/{{ item.name }}.qcow2"
+        state: absent
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host if item.libvirt_host is defined else 'localhost' }}"
+
+    - name: Remove cloud-init iso image(s)
+      file:
+        path: "{{ libvirt_default_image_path }}/{{ item.name }}-cloud-init.iso"
+        state: absent
+      with_items: "{{ molecule_yml.platforms }}"
```

Comment:
Do we want to keep `with_items` or do we want to use `loop`?

Reply:
with_items has been changed for loop.

Reply:
`with_items` has been changed for `loop`.